Drop params support for OpenAI and Azure

S1LV3RJ1NX opened the pull request:
## Title
Add support for OpenAI image generation models by introducing `GPTImageConfig` and `DallEImageConfig` classes. Updated `get_optional_params_image_gen` function to handle new configurations.

`response_format` parameter and any other non-compatible params should be ommited when `drop_params=True` else raise an Exception

## Relevant issues
#10501 

## Type

🆕 New Feature

S1LV3RJ1NX committed:
Add support for OpenAI image generation models by introducing GPTImageConfig and DallEImageConfig classes. Updated get_optional_params_image_gen function to handle new configurations.

S1LV3RJ1NX committed:
Updated image name search

ishaan-jaff committed:
[Fix + Refactor] Trigger Soft Budget Webhooks When Key Crosses Threshold (#10491)

* fix slack alerting with webhooks

* emit correct event group/entity on webhooks

* refactor to use a common class of alerts with abc methods

* fixes for tests

* refactor to use a common class of alerts with abc methods

* Send a budget alert on slack or webhook

* unit test slack alerting

* fix code qa

ishaan-jaff committed:
[Bug Fix] Ensure Web Search / File Search cost are only added when the response includes the too call (#10476)

* only apply cost if response includes annotations for url/file

* only apply cost if response includes annotations for url/file

* testing fix tool cost tracking

* fix response_includes_annotation_type

ishaan-jaff committed:
test_get_cost_for_built_in_tools_file_search

Commit:
fix(token_counter.py): handle empty lists

Commit:
fix: fix linting errors

S1LV3RJ1NX committed:
Refactor budget assertions in tests to improve clarity and accuracy. Updated remaining hours check to ensure positive values and adjusted budget reset time validation for better range checks. (#10500)

Commit:
bump: version 1.67.5 → 1.67.6

Commit:
build: update litellm version

S1LV3RJ1NX commented:
For OpenAI: `gpt-image-1`

When `drop_params=False`
![image](https://github.com/user-attachments/assets/cb60e607-6ff9-4363-a093-d0171fb4a6f3)


When `drop_params=True`
![image](https://github.com/user-attachments/assets/2f37315e-977d-4690-a246-fa3988cd8d99)

Commit:
test: bump limit on flaky load test

ishaan-jaff committed:
[Feat] KnowledgeBase/Vector Store - Log `StandardLoggingVectorStoreRequest` for requests made when a vector store is used  (#10509)

* ensure vector store results are logged in SLP

* fix tests

* fix tests with vector_store_request_metadata

* fix linting

ishaan-jaff committed:
Don't depend on uvloop on windows (#10060) (#10483)

Co-authored-by: Dave Cunningham <sparkprime@gmail.com>

Elijas committed:
fix: PydanticDeprecatedSince20: Support for class-based `config` is deprecated, use ConfigDict instead. Deprecated in Pydantic V2.0 to be removed in V3.0. (#9372)

ishaan-jaff committed:
[Feat] Show Vector Store / KB Request on LiteLLM Logs Page (#10514)

* ensure vector store results are logged in SLP

* fix tests

* fix tests with vector_store_request_metadata

* fix linting

* track duration of vector store, only log content when user opts into it

* working vector store viewer

* fix custom llm provider - Vector Store Requests

* fix vector store viewer

* fix logging redacted vector searches

* testing for storing KB queries in DB

msabramo committed:
Fix pytest event loop warning (#10512)

Fixes: GH-9641

This is a Cursor-generated fix for the following warning from pytest:

```
litellm/caching/llm_caching_handler.py:17
  /Users/abramowi/Code/OpenSource/litellm/litellm/caching/llm_caching_handler.py:17: DeprecationWarning: There is no current event loop
    event_loop = asyncio.get_event_loop()
```

Cursor prompt:

  Fix this pytest warning
  ```
  litellm/caching/llm_caching_handler.py:17
    /Users/abramowi/Code/OpenSource/litellm/litellm/caching/llm_caching_handler.py:17: DeprecationWarning: There is no current event loop
      event_loop = asyncio.get_event_loop()
  ```

Fixes https://github.com/BerriAI/litellm/issues/9641

Commit:
UI - fix adding vertex models with reusable credentials + fix pagination on keys table + fix showing org budgets on table (#10528)

* feat(provider_specific_fields.tsx): Use correct form instance

Fixes https://github.com/BerriAI/litellm/issues/10115

* Fix broken pagination by correctly passing page and pageSize to keyListCall (#10498)

* [Organization] Include litellm_budget_table in /organization/list response (#10488)

* join litellm_budget_table in /organization/list endpoint

* update test

---------

Co-authored-by: tanjiro <56165694+NANDINI-star@users.noreply.github.com>

Commit:
Playwright test for team admin (#10366) (#10470)

* Playwright test for team admin (#10366)

* Create team_admin.spec.ts

* add todo

* modify test and timeout (#10525)

---------

Co-authored-by: tanjiro <56165694+NANDINI-star@users.noreply.github.com>

ishaan-jaff committed:
[QA] Bedrock Vector Stores Integration - Allow using with registry + in OpenAI API spec with tools (#10516)

* refactor KB implementation to use central registry

* allow passing tools when making KB calls

* test fixes

* linting fix

* fix kb tests

* QA for KB stored in DB

* fix, use litellm_credential_name when adding KB on litellm UI

* QA list endpoint vector stores

* allow using UI creds with KBs

Commit:
UI - allow reassigning team to other org (#10527)

* feat(team_info.tsx): allow user to reassign team to another org

* style(team_info.tsx): fix org id styling

* feat(team_endpoints.py): add validation check before migrating team to another org

ensure model access, budgets and membership is respected

* fix(team_endpoints.py): update model migration validation to check if org has 'all-proxy-models' access

* fix(organization_view.tsx): show teams belonging to org

* feat(team_endpoints.py): handle wildcard model check on org migration

* fix(team_endpoints.py): nest router check

* test: update testing - use model with higher quota

* build: update poetry lock

NANDINI-star committed:
fix edit credentials modal (#10519)

ishaan-jaff committed:
bump litellm proxy extras

ishaan-jaff committed:
fix test_anthropic_messages_litellm_router_streaming_with_logging

iwamot committed:
Add Amazon Nova Premier (#10520)

user202729 committed:
Update contributing-to-documentation documentation to include pnpm guide (#10523)

Commit:
Add customer + model per key level multi-instance tpm/rpm limiting (#10518)

* fix(redis_cache.py): handle multiple event loops

* fix(parallel_request_limiter_v2.py): add customer tpm limiting

* fix(parallel_request_limiter.py): add customer rpm limiting

* fix(parallel_request_limiter_v2.py): add model per key + customer tpm/rpm limiting

* fix(parallel_request_limiter_v2.py): make error more informative

* fix: fix ruff error

* fix: generate new poetry lock

Commit:
build: update version + new ui build

ishaan-jaff committed:
[Docs] Using LiteLLM with vector stores / knowledge bases (#10534)

* docs vector store registry

* img for kbs

* docs kb with litellm

* fix code qa

ishaan-jaff committed:
fix typo (#10535)

ishaan-jaff committed:
docs add section on bedrock provider to show how to use KBs

ishaan-jaff committed:
docs VectorStoreRegistry

ishaan-jaff committed:
ui new build

ishaan-jaff committed:
fix linting error

Commit:
Litellm stable release notes 05 03 2025 (#10536)

* build(release_cycle.md): document bar for minor vs. patch updates

* docs(index.md): initial changelog doc

* docs(index.md): update llama docs

* docs(index.md): add docs for llm api endpoints + spend tracking/budget improvements

* docs: more doc cleanup

* docs(index.md): more doc cleanup

* docs(index.md): final doc cleanup

ishaan-jaff committed:
add perplexity/sonar-deep-research (#10537)

ishaan-jaff committed:
text fix vertex deprecated a model

ishaan-jaff committed:
text fix redis cache

ishaan-jaff committed:
[Contributor PR] Support Llama-api as an LLM provider (#10451) (#10538)

* Support Llama-api as an LLM provider (#10451)

* init: support llama-api as a llm provider

* docs: fix endpoint url

* fix: rename meta dir to meta-llama

* docs: add meta-llama info

* fix: mv LlamaAPIConfig under chat directory

* feat: add LlamaAPIConfig in ProviderConfigManager

* fix: provider_config from ProviderConfigManager

* feat: add supports_tool_choice param

* fix: remove optional_params using model_info

* fix: rename meta-llama to meta_llama

* init: test for meta_llama

* fix: model names

---------

Co-authored-by: Krish Dholakia <krrishdholakia@gmail.com>

* fix file naming convention

* fix file naming convention for meta_llama

* docs meta llama api litellm

---------

Co-authored-by: Young Han <110819238+seyeong-han@users.noreply.github.com>
Co-authored-by: Krish Dholakia <krrishdholakia@gmail.com>

ishaan-jaff committed:
fixes for web search tests

Commit:
test: add timeout on keys page

Commit:
test: reduce timeout

Commit:
test: comment out new ui test - needs more investigation

Commit:
fix(model_management_endpoints.py): allow team admin to update model … (#10539)

* fix(model_management_endpoints.py): allow team admin to update model via `/model/{model_id}/update` route

Fixes ui regression where team admin could not modify their own models

* fix(provider_specific_fields.tsx): style fix

* fix(table.tsx): allow expanding multiple rows

* fix(organization_endpoints.py): more robust check if user can give org model access

handle when user has models=["all-proxy-models"]

* fix(organization_endpoints.py): enable proxy admin with 'all-proxy-model' access to create new org with specific models

Fixes LIT-135

* fix: fix linting error

* fix: fix ui linting error

* fix(index.tsx): fix linting errors

ishaan-jaff committed:
test_openai_responses_api_web_search_cost_tracking

ishaan-jaff committed:
add meta llma api to release notes

ishaan-jaff committed:
[Bug Fix] UnicodeDecodeError: 'charmap' on Windows during litellm import  (#10542)

* test using litellm on windows

* fix using litellm on windows

* test using litellm on windows

* fix change on ci/cd

* fix test on ci/cd

Commit:
fix(converse_transformation.py): handle meta llama tool call response (#10541)

* fix(converse_transformation.py): handle meta llama tool call response

Fixes issue where bedrock meta llama would return tool call response as content str

* test(test_converse_transformation.py): add unit testing for new function

* fix: fix linting error

* fix: fix linting error

ishaan-jaff committed:
ui new build

Commit:
docs(meta_llama.md): clarify meta llama works on responses api

ishaan-jaff committed:
add vector stores to release note

Commit:
docs(index.md): add key highlights to docs

zoltan-ongithub committed:
Github: Increase timeout of litellm tests (#10568)

seyeong-han committed:
[Docs] Change llama-api link for litellm (#10556)

* docs: change llama-api link for litellm

* docs: init supported models table

ishaan-jaff commented on litellm/llms/openai/image/gpt_image_transformation.py:
```diff
@@ -0,0 +1,79 @@
+        return False
+    
+    @classmethod
+    def map_openai_params(cls, non_default_params: dict, optional_params: dict) -> dict:
```
shouldn't it be this 

```
supported_openai_params = self.get_supported_openai_params(model)
        for param, value in non_default_params.items():
            if param in supported_openai_params:
                optional_params[param] = value
        return optional_params
```

ishaan-jaff commented on litellm/llms/openai/image/dall_e_image_transformation.py:
```diff
@@ -0,0 +1,80 @@
+        return False
+    
+    @classmethod
+    def map_openai_params(cls, non_default_params: dict, optional_params: dict) -> dict:
```
shouldn't it be this ?

```
supported_openai_params = self.get_supported_openai_params(model)
        for param, value in non_default_params.items():
            if param in supported_openai_params:
                optional_params[param] = value
        return optional_params
```

ishaan-jaff reviewed (changes requested):
reviewed

ishaan-jaff committed:
[Feat] v2 Custom Logger API Endpoints (#10575)

* fixes for generic api logger

* tests for generic api logger

* test_generic_api_callback_multiple_logs

* allow health checking generic api endpoints

* docs generic api endpoint for logging

* allow setting headers for generic api  callback

* fix for test_init_custom_logger_compatible_class_as_callback

* fix linting

ishaan-jaff committed:
fix logging with JSON logs

ishaan-jaff committed:
Revert "fix logging with JSON logs"

This reverts commit 1109a6ea6eda63a9bf10c42fa4b0bfb5a3f6145c.

ishaan-jaff committed:
[Bug fix] JSON logs - Ensure only 1 log is emitted (previously duplicate json logs were getting emitted) (#10580)

* fix logging with JSON logs

* test: test_initialize_loggers_with_handler_sets_propagate_false

Commit:
docs(index.md): add graphic

mkavinkumar1 committed:
Update gemini-2.5-pro-exp-03-25 max_tokens to 65,535 (#10548)

Fixes:
Error generating conversation title: litellm.BadRequestError: VertexAIException BadRequestError - {
  "error": {
    "code": 400,
    "message": "Unable to submit request because it has a maxOutputTokens value of 65536 but the supported range is from 1 (inclusive) to 65536 (exclusive). Update the value and try again.",
    "status": "INVALID_ARGUMENT"
  }
}

S1LV3RJ1NX committed:
Refactor map_openai_params method in DallEImageConfig and GPTImageConfig classes to selectively update optional parameters based on supported OpenAI parameters.

thomelane committed:
Update instructor.md (#10549)

Simplified examples

frankzye committed:
fix issue when databrick use external model, the delta could be empty content (#10540)

ishaan-jaff committed:
Add `litellm-proxy` CLI (#10478) (#10578)

* Add `litellm-proxy` CLI (#10478)

* First cut at a Python client module for proxy

* Add UnauthorizedError + add_model method

* Add delete_model method

* Add example model_id to delete_model docstring

* Make delete_model raise NotFoundError

* Add get_model

* Add get_all_model_info

* Rename models.list_models to models.list

* Rename models.get_all_model_info to models.info

* Move ModelsManagementClient.get_all_model_group_info to ModelGroupsManagementClient.info

* Rename get_model to get

* Rename add_model to new

* Rename delete_model to delete

* In client classes, rename base_url attribute to _base_url and api_key attribute to _api_key

* Add ModelsManagementClient.updae method

* Add client.chat.completions (ChatClient)

* ruff format litellm/proxy/client

* ruff format tests/litellm/proxy/client/*.py

* Add latest changes

* Rename KeysManagementClient.create to KeysManagementClient.generate

* Add new parameters to KeysManagementClient.generate

* Add CredentialsManagementClient

* Remove api_key parameter from KeysManagementClient.generate

* Fix lint errors

* Add litellm/proxy/client/README.md

* README.md: Remove api_key param to client.keys.generate

* Fix mypy errors

* First cut at litellm-proxy cli

* Add test for `litellm-proxy models list`

* Nicer get_models_info

* get_models_info: --columns option

* Use format_timestamp in list_models

* ruff format litellm/proxy/client

* Simpler JSON printing with rich.print_json

* Move models-related commands to separate file

From `cli.py` to `groups/models.py`

* Improve directory structure

* Cleanup cli/groups/models.py - esp. usage of rich

* Refactoring

* Refactor mocking in cli/test_main.py

* Dedup models commands tests

* Update poetry.lock

* Fix mypy errors

* ruff format litellm/proxy/client/cli

* ruff format tests/litellm/proxy/client/*.py

* Fix timezone issue in test_models_list_table_format

* Add cli/README.md

* Small README.md tweaks

* README.md enhancements

* Add credentials commands

* Add chat commands

* Add http commands

* ruff format litellm/proxy/client/cli

* Fix lint errors in credentials and http commands

* json => json_lib

* test-key => sk-test-key

* Mock HTTP responses so http command tests pass

* Fix mypy error in credentials.py

* bump: version 1.67.5 → 1.67.6

* build: update litellm version

* cli/main.py: show_envvar=True

* Increase test job timeout to 8 minutes

because it looks like maybe the job is getting canceled because it takes
too long with the additional tests?

This probably could be reverted once #10484 is merged, since that speeds
up pytest runs greatly.

* Add keys functionality to library/CLI

* Add info about keys commands to litellm/proxy/client/cli/README.md

* Move Model Information section in CLI README

* Make Model Information a level 4 heading

* Move rich to extras

as suggested by @ishaan-jaff

---------

Co-authored-by: Krrish Dholakia <krrishdholakia@gmail.com>

* pin rich=13.7.1

---------

Co-authored-by: Marc Abramowitz <abramowi@adobe.com>
Co-authored-by: Krrish Dholakia <krrishdholakia@gmail.com>

ishaan-jaff committed:
bump to 1.68.1

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10504) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10504) before we can accept your contribution.<br/>**11** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: ishaan-jaff<br/>:white_check_mark: S1LV3RJ1NX<br/>:white_check_mark: msabramo<br/>:white_check_mark: krrishdholakia<br/>:white_check_mark: iwamot<br/>:white_check_mark: user202729<br/>:white_check_mark: zoltan-ongithub<br/>:white_check_mark: mkavinkumar1<br/>:white_check_mark: frankzye<br/>:white_check_mark: thomelane<br/>:white_check_mark: NANDINI-star<br/>:x: Elijas<br/>:x: seyeong-han<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10504) it.</sub>

S1LV3RJ1NX commented:
Closing this PR, as log of changes got pulled will open a new one.
